### PR TITLE
[Dynamic Instrumentation] Normalize redaction keywords + add missing keywords

### PR DESCRIPTION
**Pull request by shurivich**
## Summary of changes
Normalize redaction keywords
Add missing keywords
Add unit tests

## Reason for change
PII redaction was not working as expected because certain keywords weren't normalized. This update fix that issue. Additionally, after this functionality was initially introduced, some keywords were added; this update now includes them.
Also added some unit tests.

### CODE REVIEW

File: tracer/src/Datadog.Trace/Debugger/Snapshots/Redaction.cs [hunk 1]
```diff
@@ -78,125 +78,114 @@ internal static class Redaction
 
         private static HashSet<string> _redactKeywords = new()
```
Comment (andrewlock) [inline tracer/src/Datadog.Trace/Debugger/Snapshots/Redaction.cs:79]: Would it be preferable to make this a case-insensitive comparer?
```suggestion
        private static HashSet<string> _redactKeywords = new(StringComparer.OrdinalIgnoreCase)
```

**Comment by andrewlock** [comment]
## Execution-Time Benchmarks Report :stopwatch:

Execution-time results for samples comparing the following branches/commits:
- [This PR (5350)](https://github.com/DataDog/dd-trace-dotnet/tree/85f309cb97f956beaaaeea6566706be3524d7b9c)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/cdfb80ca4f70bd9dbf789ae02fe4509d79615bd1)

Execution-time benchmarks measure the whole time it takes to execute a program. And are intended to measure the one-off costs. Cases where the execution time results for the PR are worse than latest master results are shown in **red**. The following thresholds were used for comparing the execution times:
* Welch test with statistical test for significance of **5%**
* Only results indicating a difference greater than **5%** and **5 ms** are considered.


Note that these results are based on a _single_ point-in-time result for each branch. For full results, see the [dashboard](https://ddstaging.datadoghq.com/dashboard/4qn-6fi-54p/apm-net-execution-time-benchmarks).

Graphs show the p99 interval based on the mean and StdDev of the test run, as well as the mean value of the run (shown as a diamond below the graph).
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5350) - mean (74ms)  : 65, 83
     .   : milestone, 74,
    master - mean (73ms)  : 65, 81
     .   : milestone, 73,

    section CallTarget+Inlining+NGEN
    This PR (5350) - mean (1,006ms)  : 982, 1030
     .   : milestone, 1006,
    master - mean (1,004ms)  : 981, 1026
     .   : milestone, 1004,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5350) - mean (113ms)  : 107, 119
     .   : milestone, 113,
    master - mean (111ms)  : 108, 115
     .   : milestone, 111,

    section CallTarget+Inlining+NGEN
    This PR (5350) - mean (726ms)  : 705, 747
     .   : milestone, 726,
    master - mean (724ms)  : 700, 749
     .   : milestone, 724,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5350) - mean (95ms)  : 92, 97
     .   : milestone, 95,
    master - mean (94ms)  : 91, 97
     .   : milestone, 94,

    section CallTarget+Inlining+NGEN
    This PR (5350) - mean (679ms)  : 653, 705
     .   : milestone, 679,
    master - mean (679ms)  : 656, 702
     .   : milestone, 679,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5350) - mean (188ms)  : 185, 191
     .   : milestone, 188,
    master - mean (188ms)  : 183, 193
     .   : milestone, 188,

    section CallTarget+Inlining+NGEN
    This PR (5350) - mean (1,078ms)  : 1056, 1100
     .   : milestone, 1078,
    master - mean (1,071ms)  : 1052, 1090
     .   : milestone, 1071,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5350) - mean (271ms)  : 268, 274
     .   : milestone, 271,
    master - mean (270ms)  : 263, 276
     .   : milestone, 270,

    section CallTarget+Inlining+NGEN
    This PR (5350) - mean (868ms)  : 842, 893
     .   : milestone, 868,
    master - mean (866ms)  : 844, 888
     .   : milestone, 866,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5350) - mean (262ms)  : 257, 266
     .   : milestone, 262,
    master - mean (260ms)  : 254, 266
     .   : milestone, 260,

    section CallTarget+Inlining+NGEN
    This PR (5350) - mean (855ms)  : 829, 882
     .   : milestone, 855,
    master - mean (852ms)  : 826, 877
     .   : milestone, 852,

```

File: tracer/src/Datadog.Trace/Debugger/Snapshots/Redaction.cs (hunk 1 above)
Comment (andrewlock) [reply tracer/src/Datadog.Trace/Debugger/Snapshots/Redaction.cs:79]: In which case, you can use C#12 collection literals! 😄

Comment (shurivich) [reply tracer/src/Datadog.Trace/Debugger/Snapshots/Redaction.cs:79]: I'm not sure I understand, can you give me a code suggestion, please?

**Comment by andrewlock** [comment]
## Throughput/Crank Report:zap:

Throughput results for AspNetCoreSimpleController comparing the following branches/commits:
- [This PR (5350)](https://github.com/DataDog/dd-trace-dotnet/tree/85f309cb97f956beaaaeea6566706be3524d7b9c)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/cdfb80ca4f70bd9dbf789ae02fe4509d79615bd1)
- [benchmarks/2.9.0](https://github.com/DataDog/dd-trace-dotnet/tree/89ae3f4ee7d85b824badda5873166707c9e560b4)

Cases where throughput results for the PR are worse than latest master (5% drop or greater), results are shown in **red**.

Note that these results are based on a _single_ point-in-time result for each branch. For full results, see [one](https://ddstaging.datadoghq.com/dashboard/fnz-afi-c2i/apm-net-crank) of the [many](https://ddstaging.datadoghq.com/dashboard/c5g-i7d-kcy/ciapp-apm-net-throughput-tests), [many](https://ddstaging.datadoghq.com/dashboard/uxh-m8j-qhi/ciapp-apm-net-throughput-tests-kevin) dashboards!

```mermaid
gantt
    title Throughput Linux x64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (5350) (11.599M)   : 0, 11598773
    master (11.234M)   : 0, 11234211
    benchmarks/2.9.0 (11.502M)   : 0, 11502400

    section Automatic
    This PR (5350) (7.992M)   : 0, 7992150
    master (7.892M)   : 0, 7892025
    benchmarks/2.9.0 (8.282M)   : 0, 8282092

    section Trace stats
    This PR (5350) (8.215M)   : 0, 8215130
    master (8.128M)   : 0, 8128046

    section Manual
    This PR (5350) (10.004M)   : 0, 10003783
    master (9.998M)   : 0, 9997703

    section Manual + Automatic
    This PR (5350) (7.562M)   : 0, 7561919
    master (7.509M)   : 0, 7508673

    section Version Conflict
    This PR (5350) (6.745M)   : 0, 6745413
    master (6.752M)   : 0, 6751551

```
```mermaid
gantt
    title Throughput Linux arm64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (5350) (9.658M)   : 0, 9657745
    master (9.719M)   : 0, 9719411
    benchmarks/2.9.0 (9.757M)   : 0, 9756597

    section Automatic
    This PR (5350) (4.734M)   : crit ,0, 4734157
    master (6.497M)   : 0, 6496614

    section Trace stats
    This PR (5350) (6.907M)   : 0, 6907427
    master (6.785M)   : 0, 6785156

    section Manual
    This PR (5350) (8.139M)   : 0, 8139463
    master (8.060M)   : 0, 8060205

    section Manual + Automatic
    This PR (5350) (6.143M)   : 0, 6142795
    master (5.959M)   : 0, 5958606

    section Version Conflict
    This PR (5350) (5.626M)   : 0, 5625942
    master (5.660M)   : 0, 5660029

```
```mermaid
gantt
    title Throughput Windows x64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (5350) (9.674M)   : 0, 9673633
    master (10.074M)   : 0, 10073856
    benchmarks/2.9.0 (9.814M)   : 0, 9813729

    section Automatic
    This PR (5350) (6.853M)   : 0, 6853231
    master (7.026M)   : 0, 7026117
    benchmarks/2.9.0 (7.278M)   : 0, 7278101

    section Trace stats
    This PR (5350) (7.159M)   : 0, 7158903
    master (7.281M)   : 0, 7281475

    section Manual
    This PR (5350) (8.372M)   : 0, 8371851
    master (8.613M)   : 0, 8613073

    section Manual + Automatic
    This PR (5350) (6.563M)   : 0, 6562960
    master (6.856M)   : 0, 6855748

    section Version Conflict
    This PR (5350) (5.951M)   : 0, 5950622
    master (6.081M)   : 0, 6080626

```

**Review by OmerRaviv — APPROVED** [review]
LGTM, great job on the test!

**Review by GreenMatan — APPROVED** [review]
LGTM

File: tracer/src/Datadog.Trace/Debugger/Snapshots/Redaction.cs (hunk 1 above)
Comment (shurivich) [reply tracer/src/Datadog.Trace/Debugger/Snapshots/Redaction.cs:79]: I understand now, will correct it next PR I will on `redaction` feature

**Pull request merged**